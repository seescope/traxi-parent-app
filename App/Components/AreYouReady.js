/* eslint global-require: "off" */

import React from 'react';
import { View, Text, Image } from 'react-native';
import { Actions } from 'react-native-router-flux';
import * as Animatable from 'react-native-animatable';

import Background from '../Components/Background';
import Button from '../Components/Button';
import Spacing from '../Components/Spacing';
import { WHITE, TRANSPARENT } from '../Constants/Colours';
import loginWithMethod from '../Actions/LoginWithMethod';
import { logError } from '../Utils';

export const handleError = error => {
  logError(`Error signing up: ${error.message}`);
  return null;
};

export const beginSetup = () => dispatch =>
  dispatch(loginWithMethod('anonymous'))
  .then(() => Actions.createKid())
  .catch(handleError);

const logoStyle = {
  backgroundColor: TRANSPARENT,
  color: WHITE,
  fontSize: 32,
  textAlign: 'center',
  fontFamily: 'Raleway-ExtraBold',
  marginHorizontal: 16,
  marginBottom: 16,
};

const containerStyle = {
  flex: 1,
  alignItems: 'center',
  paddingHorizontal: 32,
};

const SplashScreen = () =>
  <Background style={containerStyle}>
    <Animatable.Image
      animation="bounceInDown"
      source={require('../Images/trail_preview.png')}
    />

    <Spacing height={32} />

    <Animatable.Text
      animation="bounceIn"
      delay={1000}
      style={logoStyle}
    >
      Start seeing what your kids are doing.
    </Animatable.Text>

    <Spacing height={32} />

    <Animatable.View
      animation="bounceInUp"
      delay={2000}
    >
      <Button onPress={() => {}}>Show me how</Button>
    </Animatable.View>
  </Background>;


export default SplashScreen;
