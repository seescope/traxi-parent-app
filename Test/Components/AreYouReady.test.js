import React from 'react';
import renderer from 'react-test-renderer';

import AreYouReady from '../../App/Components/AreYouReady';

it('renders the <AreYouReady> component', () => {
  const tree = renderer.create(
    <AreYouReady />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
