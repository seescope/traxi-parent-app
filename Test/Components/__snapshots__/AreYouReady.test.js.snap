exports[`test renders the <AreYouReady> component 1`] = `
<Image
  resizeMode="cover"
  source={1}
  style={
    Object {
      "flex": 1,
      "width": 750,
    }
  }>
  <View
    style={
      Object {
        "flex": 7,
      }
    } />
  <View
    style={
      Object {
        "flex": 3,
        "paddingHorizontal": 32,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#FFF",
          "fontFamily": "Raleway-ExtraBold",
          "fontSize": 26,
          "textAlign": "center",
        }
      }>
      [missing \"en.areYouReady.header\" translation]
    </Text>
    <View
      style={
        Object {
          "height": 32,
        }
      } />
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "Raleway-Regular",
              "fontSize": 20,
            }
          }>
          [missing \"en.areYouReady.notReadyYet\" translation]
           
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "Raleway-Regular",
              "fontSize": 20,
            }
          }>
          [missing \"en.areYouReady.ready\" translation]
           
        </Text>
      </View>
    </View>
  </View>
</Image>
`;
